### PR TITLE
[4782] Add <- and -> (forward and back) buttons to the URL/internal browser in Preview Toolbar

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/4782

### CODE REVIEW

**Review by rart — APPROVED** [review]

**Review by rart — APPROVED** [review]

**Comment by sumerjabri** [comment]
@rart I'll wait for the final updates we did today to make it in (I wasn't able to find it in this PR).

File: ui/app/src/state/reducers/previewNavigation.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    // What's shown to the user across the board (url, address bar, etc)
+    currentUrlPath: '',
+    // The src of the iframe
+    currentFullUrl: previewLanding,
+    historyBackStack: [],
+    historyForwardStack: [],
+    historyNavigationType: null
+  },
+  {
+    [checkInGuest.type]: (state, { payload }) => {
```
Comment (rart) [inline ui/app/src/state/reducers/previewNavigation.ts:43]: Please add these type of comments to each action
```suggestion
    // A page is being visited or reloaded
    // - Push the url into the back stack (if not already pushed)
    // - Clear the forward stack (if not already pushed)
    // - Update the current url
    [checkInGuest.type]: (state, { payload }) => {
```

File: ui/guest/src/react/GuestProxy.tsx [hunk 2]
```diff
@@ -307,9 +307,9 @@ export function GuestProxy() {
               take(1)
             )
             .subscribe(function({ payload }) {
-              const { modelId, fieldId, targetIndex, currentUrl } = payload;
+              const { modelId, fieldId, targetIndex, currentFullUrl } = payload;
               let ifrm = document.createElement('iframe');
-              ifrm.setAttribute('src', `${currentUrl}`);
+              ifrm.setAttribute('src', `${currentFullUrl}`);
```
Comment (rart) [inline ui/guest/src/react/GuestProxy.tsx]:
```suggestion
              ifrm.setAttribute('src', currentFullUrl);
```

File: ui/app/src/state/reducers/previewNavigation.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+      if (payload.nextUrl !== nextState.currentUrlPath) {
+        nextState = {
+          ...nextState,
+          currentUrlPath: payload.nextUrl
+        };
+      }
+
+      return nextState;
+    }
```
Comment (rart) [inline ui/app/src/state/reducers/previewNavigation.ts]:
```suggestion
    [changeSite.type]: (state, { payload }) => ({
      ...state,
      historyBackStack: [],
      historyForwardStack: [],
      historyNavigationType: null,
      currentUrlPath: payload.nextUrl ?? state.currentUrlPath
    })
```

File: ui/app/src/modules/Preview/PreviewConcierge.tsx [hunk 4]
```diff
@@ -629,7 +631,7 @@ export function PreviewConcierge(props: any) {
 
               hostToGuest$.next({
                 type: INSERT_OPERATION_COMPLETE,
-                payload: { ...payload, currentUrl }
+                payload: { ...payload, currentFullUrl: guestBase + currentUrlPath }
```
Comment (rart) [inline ui/app/src/modules/Preview/PreviewConcierge.tsx]:
```suggestion
                payload: { ...payload, currentFullUrl: ˋ${guestBase}${currentUrlPath}ˋ }
```

File: ui/app/src/modules/Preview/PreviewConcierge.tsx [hunk 5]
```diff
@@ -588,7 +590,7 @@ export function PreviewConcierge(props: any) {
...
```
Comment (rart) [inline ui/app/src/modules/Preview/PreviewConcierge.tsx]:
```suggestion
                payload: { ...payload, currentFullUrl: ˋ${guestBase}${currentUrlPath}ˋ }
```

File: ui/app/src/modules/Preview/Host.tsx [hunk 6]
```diff
@@ -192,7 +195,7 @@ export default function Host() {
       className={clsx(classes.hostContainer, { [classes.shift]: showToolsPanel })}
     >
       <HostUI
-        url={currentUrl}
+        url={currentUrlPath === '' ? previewLandingBase : guestBase + currentUrlPath}
```
Comment (rart) [inline ui/app/src/modules/Preview/Host.tsx]:
```suggestion
        url={currentUrlPath === '' ? previewLandingBase : ˋ${guestBase}${currentUrlPath}ˋ}
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/state/reducers/previewNavigation.ts (hunk 1 above)
Comment (rart) [reply ui/app/src/state/reducers/previewNavigation.ts:43]: Still missing the comments for the rest of the actions

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**